internal/tools/generator-schema-snapshot/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	. "github.com/dave/jennifer/jen"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/provider"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+const (
+	SchemaPath = "github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		log.Fatal("Usage: generator-schema-snapshot <reource_type>")
+	}
+	rt := os.Args[1]
+	res, ok := provider.AzureProvider().ResourcesMap[rt]
+	if !ok {
+		log.Fatalf("unknown resource type %q", rt)
+	}
+
+	f := NewFile("main")
+	f.ImportName("github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk", "")
+
+	f.Var().Id("_").Op("=").Add(SchemaMap(res.Schema))
+
+	fmt.Printf("%#v", f)
+}
+
+func ResourceValue(res *pluginsdk.Resource) Dict {
+	return Dict{
+		Id("Schema"): SchemaMap(res.Schema),
+	}
+}
+
+func SchemaMap(m map[string]*pluginsdk.Schema) *Statement {
+	dict := Dict{}
+	for k, v := range m {
+		dict[Lit(k)] = Values(SchemaValue(v))
+	}
+	return Map(String()).Op("*").Qual(SchemaPath, "Schema").Values(dict)
+}
+
+func SchemaValue(sch *pluginsdk.Schema) Dict {
+	out := Dict{}
+
+	var t Code
+	switch sch.Type {
+	case pluginsdk.TypeBool:
+		t = Qual(SchemaPath, "TypeBool")
+	case pluginsdk.TypeInt:
+		t = Qual(SchemaPath, "TypeInt")
+	case pluginsdk.TypeFloat:
+		t = Qual(SchemaPath, "TypeFloat")
+	case pluginsdk.TypeString:
+		t = Qual(SchemaPath, "TypeString")
+	case pluginsdk.TypeList:
+		t = Qual(SchemaPath, "TypeList")
+	case pluginsdk.TypeMap:
+		t = Qual(SchemaPath, "TypeMap")
+	case pluginsdk.TypeSet:
+		t = Qual(SchemaPath, "TypeSet")
```

Comment:
This'll probably also need to account for/placeholder the custom hash functions too?